_studio/mfx_lib/shared/src/mfx_mpeg2_encode_vaapi.cpp

```diff
@@ -961,14 +961,17 @@ mfxStatus VAAPIEncoder::FillMiscParameterBuffer(ExecuteBuffers* pExecuteBuffers)
     sts = CheckAndDestroyVAbuffer(m_vaDisplay, m_miscParamBrcId);
     MFX_CHECK_STS(sts);
 
-    vaSts = vaCreateBuffer(m_vaDisplay,
-        m_vaContextEncode,
-        VAEncMiscParameterBufferType,
-        sizeof(VAEncMiscParameterRateControl) + sizeof(VAEncMiscParameterBuffer),
-        1,
-        m_pMiscParamsBrc,
-        &m_miscParamBrcId);
-    MFX_CHECK_WITH_ASSERT(VA_STATUS_SUCCESS == vaSts, MFX_ERR_DEVICE_FAILED);
+    if (pExecuteBuffers->m_sps.RateControlMethod == MFX_RATECONTROL_VBR)
+    {
+        vaSts = vaCreateBuffer(m_vaDisplay,
+            m_vaContextEncode,
+            VAEncMiscParameterBufferType,
+            sizeof(VAEncMiscParameterRateControl) + sizeof(VAEncMiscParameterBuffer),
+            1,
+            m_pMiscParamsBrc,
+            &m_miscParamBrcId);
+        MFX_CHECK_WITH_ASSERT(VA_STATUS_SUCCESS == vaSts, MFX_ERR_DEVICE_FAILED);
+    }
 
```

Comment:
@vilichev , I recall you fixed a similar issue in HEVC. If so, can you please review the PR?

Reply:
seems it was another issue (https://github.com/Intel-Media-SDK/MediaSDK/pull/2439)